File: SDKs/googleapis.com/docs/v1/go/pkg/models/shared/inlineobjectelement.go
package shared

type InlineObjectElement struct {
	InlineObjectID            *string                       `json:"inlineObjectId"`
	SuggestedDeletionIds      []string                      `json:"suggestedDeletionIds"`
	SuggestedInsertionIds     []string                      `json:"suggestedInsertionIds"`
	SuggestedTextStyleChanges map[string]SuggestedTextStyle `json:"suggestedTextStyleChanges"`
	TextStyle                 *TextStyle                    `json:"textStyle"`
}

File: SDKs/googleapis.com/docs/v1/typescript/src/sdk/models/shared/inlineobjectelement.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
import { SuggestedTextStyle } from "./suggestedtextstyle";
import { TextStyle } from "./textstyle";


// InlineObjectElement
/** 
 * A ParagraphElement that contains an InlineObject.
**/
export class InlineObjectElement extends SpeakeasyBase {
  @Metadata({ data: "json, name=inlineObjectId" })
  inlineObjectId?: string;

  @Metadata({ data: "json, name=suggestedDeletionIds" })
  suggestedDeletionIds?: string[];

  @Metadata({ data: "json, name=suggestedInsertionIds" })
  suggestedInsertionIds?: string[];

  @Metadata({ data: "json, name=suggestedTextStyleChanges", elemType: shared.SuggestedTextStyle })
  suggestedTextStyleChanges?: Map<string, SuggestedTextStyle>;

  @Metadata({ data: "json, name=textStyle" })
  textStyle?: TextStyle;
}
